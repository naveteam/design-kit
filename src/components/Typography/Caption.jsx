import React from 'react'
import styled, { css } from '@xstyled/styled-components'
import { variant } from '@xstyled/system'
import PropTypes from 'prop-types'

import { Typography } from './'

const Caption = ({ as, ...props }) => <Base forwardedAs={props.as} {...props} />

const sizeVariant = variant({
  default: 'md',
  key: 'caption',
  prop: 'variant',
  variants: {
    md: css`
      font-size: 1;
      line-height: 1;
    `,
    sm: css`
      font-size: 0;
      line-height: 0;
    `
  }
})

const Base = styled(Typography)`
  ${sizeVariant}
`

Caption.defaultProps = {
  variant: 'md',
  as: 'p',
  fontWeight: 0
}

Caption.propTypes = {
  ...Typography.propTypes,
  variant: PropTypes.oneOf(['sm', 'md'])
}

export default Caption
