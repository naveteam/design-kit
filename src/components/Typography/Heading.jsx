import React from 'react'
import styled, { css } from '@xstyled/styled-components'
import { variant, breakpoints } from '@xstyled/system'
import PropTypes from 'prop-types'

import { Typography } from './'

const Heading = props => <Base forwardedAs={props.variant} {...props} />

const tagVariant = variant({
  default: 'h1',
  key: 'heading',
  prop: 'variant',
  variants: {
    h1: breakpoints({
      xs: css`
        font-size: 7;
        line-height: 7;
      `,
      md: css`
        font-size: 8;
        line-height: 8;
      `
    }),
    h2: css`
      font-size: 6;
      line-height: 6;
    `,
    h3: css`
      font-size: 5;
      line-height: 5;
    `,
    h4: css`
      font-size: 4;
      line-height: 4;
    `
  }
})

const Base = styled(Typography)`
  ${tagVariant}
`

Heading.defaultProps = {
  variant: 'h1',
  fontWeight: 1
}

Heading.propTypes = {
  ...Typography.propTypes,
  variant: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4'])
}

export default Heading
