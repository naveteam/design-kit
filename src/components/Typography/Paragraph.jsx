import React from 'react'
import styled, { css } from '@xstyled/styled-components'
import { variant } from '@xstyled/system'
import PropTypes from 'prop-types'

import { Typography } from './'

const Paragraph = ({ as, ...props }) => <Base forwardedAs={props.as} {...props} />

const sizeVariant = variant({
  default: 'md',
  key: 'paragraph',
  prop: 'variant',
  variants: {
    md: css`
      font-size: 3;
      line-height: 3;
    `,
    sm: css`
      font-size: 2;
      line-height: 1;
    `
  }
})

const Base = styled(Typography)`
  ${sizeVariant}
`

Paragraph.defaultProps = {
  variant: 'md',
  as: 'p',
  fontWeight: 0
}

Paragraph.propTypes = {
  ...Typography.propTypes,
  variant: PropTypes.oneOf(['sm', 'md'])
}

export default Paragraph
