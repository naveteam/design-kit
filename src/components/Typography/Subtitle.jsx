import React from 'react'

import { Typography } from './'

const Subtitle = props => <Typography {...props} />

Subtitle.defaultProps = {
  as: 'p',
  fontWeight: 1,
  fontSize: 3,
  lineHeight: 3
}

Subtitle.propTypes = {
  ...Typography.propTypes
}

export default Subtitle
