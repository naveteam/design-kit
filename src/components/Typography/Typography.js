import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'

import modifiers from 'src/helpers/modifiers'

const Component = props => {
  const { as, children, ...rest } = props
  return React.createElement(as, rest, children)
}

const StyledTypography = styled(Component)`
  ${props => {
    const { theme, color, fontSize, fontFamily } = props
    return `
      ${modifiers(props)}
      font-family: ${fontFamily || 'Roboto'};
      color: ${color || theme.colors.darkGrey};
      ${fontSize ? `font-size: ${fontSize};` : ''}
    `
  }};
`

const Typography = props => <StyledTypography {...props} />

Typography.defaultProps = {
  as: 'p'
}

Typography.propTypes = {
  as: PropTypes.string,
  color: PropTypes.string,
  fontSize: PropTypes.string,
  fontFamily: PropTypes.string
}

export default Typography
