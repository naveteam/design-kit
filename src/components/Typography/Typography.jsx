import React from 'react'
import styled from '@xstyled/styled-components'
import { space, typography, color } from '@xstyled/system'
import PropTypes from 'prop-types'

const BaseComponent = ({ as, children, ...props }) => React.createElement(as, props, children)

const Typography = props => <StyledBase {...props} />

const StyledBase = styled(BaseComponent)`
  margin: 0;
  padding: 0;
  font-family: 'Open Sans', sans-serif;
  ${space}
  ${typography}
  ${color}
`

Typography.defaultProps = {
  as: 'p'
}

Typography.propTypes = {
  as: PropTypes.oneOf(['h1', 'h2', 'h3', 'h4', 'p', 'span', 'label'])
}

export default Typography
