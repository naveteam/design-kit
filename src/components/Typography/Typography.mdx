---
name: Typography
---

import { Playground, Props } from 'docz'
import Typography from './Typography'

# Typography

We can render text as any html tag

<Props of={Typography} />

<Playground>
  <Typography as='h1'>As H1</Typography>
  <Typography as='span'>As span</Typography>
  <Typography as='div'>As div</Typography>
  <Typography color='blue'>colored</Typography>
</Playground>
