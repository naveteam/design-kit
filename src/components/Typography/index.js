import Typography from './Typography'

export * from './Typography'

export { Typography }
