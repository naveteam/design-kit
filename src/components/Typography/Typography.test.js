import React from 'react'

import { renderWithTheme } from 'helpers/testTheme'

import Typography from './Typography'

describe('Typography', () => {
  it('renders without pass children', () => {
    const { container } = renderWithTheme(<Typography />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders <p> tag if no 'as'", () => {
    const { container } = renderWithTheme(<Typography>font</Typography>)

    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders <h1> element if passed as 'as'", () => {
    const { container } = renderWithTheme(<Typography as='h1'>font</Typography>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with red color', () => {
    const { container } = renderWithTheme(<Typography color='red'>font</Typography>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('renders with custom font-size of 40px', () => {
    const { container } = renderWithTheme(<Typography fontSize='40px'>font</Typography>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
