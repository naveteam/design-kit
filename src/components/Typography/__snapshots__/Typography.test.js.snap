// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders <h1> element if passed as 'as' 1`] = `
.c0 {
  font-family: Roboto;
  color: #5d5d5d;
}

<h1
  class="c0"
>
  font
</h1>
`;

exports[`Typography renders <p> tag if no 'as' 1`] = `
.c0 {
  font-family: Roboto;
  color: #5d5d5d;
}

<p
  class="c0"
>
  font
</p>
`;

exports[`Typography renders with custom font-size of 40px 1`] = `
.c0 {
  font-family: Roboto;
  color: #5d5d5d;
  font-size: 40px;
}

<p
  class="c0"
  font-size="40px"
>
  font
</p>
`;

exports[`Typography renders with red color 1`] = `
.c0 {
  font-family: Roboto;
  color: red;
}

<p
  class="c0"
  color="red"
>
  font
</p>
`;

exports[`Typography renders without pass children 1`] = `
.c0 {
  font-family: Roboto;
  color: #5d5d5d;
}

<p
  class="c0"
/>
`;
